hogben/tests/test_samples.py

```diff
@@ -0,0 +1,313 @@
+    substrate = SLD_refl1d(rho=2.047, name='Substrate')(thickness=0,
+                                                        interface=2)
+    structure = substrate | layer2 | layer1 | air
+    return Sample(structure)
+
+def mock_save_plot(fig: matplotlib.figure.Figure,
+                   save_path: str,
+                   filename: str) -> None:
+    """
+    A mocked version of the hogben.utils.save_plot method, where a lower
+    dpi is used when saving a figure
+
+    Args:
+        fig: The matplotlib figure to be plotted
+        save_path: The path where the figure will be saved
+        filename: The file name of the figure without the png extension
+    """
+    if not os.path.exists(save_path):
+        os.makedirs(save_path)
+    file_path = os.path.join(save_path, filename + '.png')
+    fig.savefig(file_path, dpi=40)
+
+def compare_sample_structure(refl1d: refl1d.model.Stack,
+                             refnx: refnx.reflect.Structure) -> None:
+    """
+    Compare the structural parameters of a refnx and refl1d model, and check
+    if the values are equal. This function is as a helper function to
+    validate that a converted refl1d/refnx sample still has the same
+    structural parameters and values as it had before conversion.
+    Args:
+        refl1d: The refl1d structure to be compared
+        refnx: The refnx structure to be compared
+
+    Returns:
+        bool: Boolean describing whether the two models have the same values
+    """
+    refl1d_params = {}
+    refnx_params = {}
+
+    # Check structure in reversed order, so it matches with refl1d
+    for component in list(reversed(refnx.structure))[1:]:
+        refnx_params[component.name] = {
+            "sld": component.sld.real.value,
+            "thick": component.thick.value,
+            "rough": component.rough.value
+        }
+    for component in refl1d.structure[1:]:
+        refl1d_params[component.name] = {
+            "sld": component.material.rho.value,
+            "thick": component.thickness.value,
+            "rough": component.interface.value
+        }
+    return refnx_params == refl1d_params
+
+@pytest.mark.parametrize('sample_class', ("refnx_sample",
+                                         "refl1d_sample"))
+def test_angle_info(sample_class, request):
+    """
+    Tests whether the angle_info function correctly calculates the Fisher
+    information, and outputs the same values as if the functions were called
+    manually.
+    """
+
+    # Get Fisher information from tested unit
+    sample = request.getfixturevalue(sample_class)
+    angle_times = [(0.7, 100, 10000), (2.0, 100, 10000)]
+    angle_info = sample.angle_info(angle_times)
+
+    # Get Fisher information directly
+    model, data = simulate(sample.structure, angle_times)
+    qs, counts, models = [data[:, 0]], [data[:, 3]], [model]
+    g = fisher(qs, sample.params, counts, models)
+
+    np.testing.assert_allclose(g, angle_info, rtol=1e-08)
```

Comment:
given the random nature of the FI calculation, 1e-8 may be a little small for a tolerance, and may cause intermittent test failures. what sort of changes are you seeing between the values?